15_tictactoe/go/grid/grid.go

```diff
@@ -1,10 +1,16 @@
 package grid
 
+// Define column and row
 type Grid struct {
-	// TODO
+	Row []int
+	Col []int
 }
 
 func (g Grid) String() string {
 	// TODO
+	// |||____
+	// |__|__|__|
+	// |__|__|__|
+	// |__|__|__|
 	return ""
 }
```

Comment:
@nobe4 Finding it a bit hard to start. Is this very basic idea going in the right direction?

Reply:
Try using a single list of ints for the cell:

```go
type Grid struct {
  Cells []int
}
```

Reply:
@nobe4 Could I get more hints for after that?

Reply:
you can do this in a single operation, using https://pkg.go.dev/fmt#Sprintf

The function should look like

```go
func (g Grid) String() string {
  return fmt.Sprintf(
    "<TODO>",
    args...
  )
}
```

Reply:
8b5035b doesn't print the grid as a grid.
Have a read at https://pkg.go.dev/fmt, especially the https://pkg.go.dev/fmt#example-package-Formats to see how you can format the output into a table.

E.g. output

```
+---+---+---+
| X |   |   |
+---+---+---+
|   | O | O |
+---+---+---+
| X |   |   |
+---+---+---+
```

Reply:
@nobe4 Another attempt. If that's not it, I think I have to try to figure out how to break it down further and manage to run and test the code first.

Reply:
@nobe4 Added string grid logic for output first with int values then with string output. I guess string output would be better?

Reply:
@nobe4 have
- visual string output
- win output

Trying to get the indexes of the free cell slice returned now.
Suppose I have to loop with range for that?
Did it before, but then I ignored the indexes and just returned the values. Seems a bit more tricky to just return the indexes...